<?xml version="1.0"?>
<!--
  Copyright (C) 2023 LEIDOS.

  Licensed under the Apache License, Version 2.0 (the "License"); you may not
  use this file except in compliance with the License. You may obtain a copy of
  the License at

  http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
  License for the specific language governing permissions and limitations under
  the License.
-->

<launch>
    <arg name="normal_operation" default="true" doc = "When this is true, `lightbar_manager` should have the highest priority in the params file"/>
    <arg name="spin_rate_hz" default="10" doc="ROS spin rate" />
    <arg name="lightbar_driver_service_name" default="set_lights" doc= "LightBarDriver's set_lights service name" />
    <arg name="guidance_state_topic_name" default= "state" doc= "GuidanceState topic name that LightBarManager listens to for changing its own state" />
    <arg name="turn_signal_topic_name" default= "turn_signal_command" doc= "TurnSignalCommand topic name that LightBarManager listens to for changing turn signal" />
    <node pkg="lightbar_manager" type="lightbar_manager_node" name="lightbar_manager">
        <param name="normal_operation" value="$(arg normal_operation)" />
        <param name="spin_rate_hz" value="$(arg spin_rate_hz)"/>
        <param name="lightbar_driver_service_name" value = "$(arg lightbar_driver_service_name)" />
        <param name="guidance_state_topic_name" value = "$(arg guidance_state_topic_name)" />
        <param name="turn_signal_topic_name" value = "$(arg turn_signal_topic_name)" />
        <rosparam command="load" file="$(find lightbar_manager)/config/params.yaml" />
    </node>
</launch>
